modules/imgproc/src/filter.simd.hpp

```diff
@@ -465,6 +465,52 @@ struct RowVec_8u32s
     bool smallValues;
 };
 
+struct RowVec_8u32f
+{
+    RowVec_8u32f() {}
+    RowVec_8u32f( const Mat& _kernel ) { kernel = _kernel; }
+
+    int operator()(const uchar* _src, uchar* _dst, int width, int cn) const
+    {
+        CV_INSTRUMENT_REGION();
+
+        int i = 0, k, _ksize = kernel.rows + kernel.cols - 1;
+        float* dst = (float*)_dst;
+        const float* _kx = kernel.ptr<float>();
+        width *= cn;
+        for( ; i < width - v_uint8::nlanes; i += v_uint8::nlanes )
+        {
+            v_float32 s0 = vx_setzero_f32();
+            v_float32 s1 = vx_setzero_f32();
+            v_float32 s2 = vx_setzero_f32();
+            v_float32 s3 = vx_setzero_f32();
+            k = 0;
+            for( ; k < _ksize ; k++ )
+            {
+                v_float32 f = vx_setall_f32(_kx[k]);
+                const uchar* src = (const uchar*)_src + i + k * cn;
+                v_uint8 vs = vx_load(src);
+                v_int16 vs_l = v_reinterpret_as_s16(v_expand_low(vs));
+                v_int16 vs_h = v_reinterpret_as_s16(v_expand_high(vs));
+                v_float32 fvs_ll = v_cvt_f32(v_expand_low(vs_l));
+                v_float32 fvs_lh = v_cvt_f32(v_expand_high(vs_l));
+                v_float32 fvs_hl = v_cvt_f32(v_expand_low(vs_h));
+                v_float32 fvs_hh = v_cvt_f32(v_expand_high(vs_h));
```

Comment:
> v_expand_low(...)
> v_expand_high(...)

OpenCV SIMD UI has `v_expand()` intrinsic to handle both calls at once.
In general, single call variant is preferable because it allows to apply better optimizations than dedicated calls.

Reply:
Thank you for looking at my code and giving me suggestion!

Code was compiled with g++ 7.5.0 with -O3 flag. Regarding the use of `v_expand()`, in my new commit I have replaced it with `vx_load_expand_q(...)` just as you suggested, so there shouldn't be the need to use `v_expand()` anymore. Please let me know if that looks good to you.